bevy_kayak_ui/src/lib.rs

```diff
@@ -36,69 +36,86 @@ pub(crate) fn to_bevy_color(color: &kayak_core::color::Color) -> Color {
 }
 
 pub fn update(world: &mut World) {
-    let bevy_context = world.remove_resource::<BevyContext>().unwrap();
-    if let Ok(mut context) = bevy_context.kayak_context.write() {
-        context.set_global_state(std::mem::take(world));
-        context.render();
-        *world = context.take_global_state::<World>().unwrap()
+    if let Some(bevy_context) = world.remove_resource::<BevyContext>() {
+        if let Ok(mut context) = bevy_context.kayak_context.write() {
+            context.set_global_state(std::mem::take(world));
+            context.render();
+            *world = context.take_global_state::<World>().unwrap()
+        }
+        world.insert_resource(bevy_context);
     }
-
-    world.insert_resource(bevy_context);
 }
 
-pub fn process_events(
-    bevy_context: Res<BevyContext>,
-    windows: Res<Windows>,
-    mut cursor_moved_events: EventReader<CursorMoved>,
-    mut mouse_button_input_events: EventReader<MouseButtonInput>,
-    mut char_input_events: EventReader<ReceivedCharacter>,
-    mut keyboard_input_events: EventReader<KeyboardInput>,
-) {
-    let window_size = if let Some(window) = windows.get_primary() {
-        Vec2::new(window.width(), window.height())
+pub fn process_events(world: &mut World) {
```

Comment:
@MrGVSV 
Maybe we should just move process_events into update?

Reply:
Seems reasonable.